src/genesys/generate.py

```diff
@@ -82,20 +85,28 @@ def main(config: GenerateConfig):
     total_samples = 0
 
     for batch_inputs, batch in dataloader:
-        responses = llm.generate(batch_inputs, sampling_params)
-        for batch_element, response in zip(batch, responses):
-            batch_element["llm_response"] = response["text"]
+        responses = llm.generate(input_ids=batch_inputs, sampling_params=sampling_params)
+        for batch_input, batch_element, response in zip(batch_inputs, batch, responses):
+            batch_element["llm_response"] = tokenizer.decode(response["token_ids"], skip_special_tokens=True)
             batch_element["response_id"] = f"{batch_element['problem_id']}_{generate_short_id()}"
             batch_element["model_name"] = config.name_model
             batch_element["generation_config"] = sampling_params
             batch_element["machine_info"] = machine_info
+            batch_element["input_ids"] = batch_input
+            batch_element["output_ids"] = response["token_ids"]
+            batch_element["proof"] = "".join(
+                build_proofs_base64(response["meta_info"]["hidden_states"], 32, 128, skip_prefill=True)
+            )
             all_results.append(batch_element)
         total_samples += len(batch)
 
         if len(all_results) >= config.sample_per_file:
             file_name = f"{config.out_file_prefix}_{uuid.uuid4()}.jsonl"
             file = os.path.join(config.path_output, file_name)
             save_batch_results(all_results, file, gcp_bucket)
+            file_sha = sha256sum(file)
+            dataloader.prime_metric.log_prime({"file_sha": file_sha, "file_name": file_name})
```

Comment:
I would refactor and instantiate the prime metric in the main script and pass it as param to the dataalodaer.

ideally the prime metric should be as simple to use as wandb

Reply:
eh should it be global then?